internal/client.go

```diff
@@ -707,6 +710,23 @@ type (
 		// Cannot be set the same time as a CronSchedule.
 		StartDelay time.Duration
 
+		// Summary - Single-line summary for this workflow execution that will appear in UI/CLI. This can be in
+		// single-line Temporal markdown format.
+		//
+		// Optional: defaults to none/empty.
+		//
+		// NOTE: Experimental
+		Summary string
+
+		// Details - General details for this workflow execution that will appear in UI/CLI. This can be in Temporal
+		// markdown format and can span multiple lines. This is a fixed value on the workflow that cannot be updated.
+		// For details that can be updated, use SetCurrentDetails within the workflow.
+		//
+		// Optional: defaults to none/empty.
+		//
+		// NOTE: Experimental
+		Details string
```

Comment:
> This is a fixed value on the workflow that cannot be updated. For details that can be updated, use SetCurrentDetails within the workflow.

It's kind of confusing `Details` works differently then search attributes and memo here. Is there a reason we decided to be incosistent here?

Reply:
Can you clarify which difference? The fact that it's not arbitrary key-value or the fact that it's a string and not a payload.

It's not arbitrary key value because it was decided in the user metadata project not to support arbitrary key value for any user metadata but make explicit fields as needed (and we can add more on the proto as needed). It's a string because the UI has an expectation on this being a string for display.

Reply:
The fact that it can't be updated

Reply:
Ah, this is by intention based on the design discussions. We want mutable information to be available via query, the metadata is meant to be immutable.

Reply:
Forgive my ignorance, but do these two `Details` appear as separate in the UI?

Reply:
I think they should, yes. We should discuss this actually.

So today we have fixed summary/details you set on a workflow. Then we have a wholly-unrelated concept of "current details" (was unable to think of a better name) that is queryable. But there has been discussion on making the query occur by default when you arrive  on the UI. There is no overlap or any relation between "fixed workflow start details" and "queryable workflow current details" except in the English term "details" we happened to have reused here.

It would be a much larger task to try and update the workflow one from the query. But one thing we can consider doing now that I am hearing we may issue the query upon page arrival, is we can default the current details to the workflow details one. The reason I am hesitant is because I don't want people to wonder why on "describe workflow" they don't get the current details.

Reply:
Yeah i think I would make a stronger distinction in the SDK API that there are two details:
* One set on start that is fixed and does not require a worker to be running
* One that is set in the workflow that DOES require the workflow to be running

I think they deserve separate names in the SDK, maybe we can call one `static workflow details` and one `dynamic workflow details`?

Reply:
Are you saying I should change the start workflow options to be `StaticSummary` and `StaticDetails`, and change the method in workflow to be `workflow.SetDynamicDetails`? One of the concerns is this now deviates from the API naming that was settled on a while back and is now implemented. I wouldn't want to have static summary not have the same prefix as static details. Maybe `StaticSummary` and `StaticDetails` and keep `workflow.SetCurrentDetails`? Or maybe `FixedSummary` and `FixedDetails`? Or maybe this can simply be documented away?

Reply:
I think we should use different names for different connects, I am not too picky on the exact names.

>. Maybe StaticSummary and StaticDetails and keep workflow.SetCurrentDetails

The works for me.

Reply:
Ok, let me bring up with the group and get group-wide consensus.

Reply:
Changed to `StaticSummary`/`StaticDetails`.